Remove cruft from ThemeMac

annevk opened the pull request:
#### 20d6a4967eff10b34ccdda0fd124f549b363afee
<pre>
Remove cruft from ThemeMac
<a href="https://bugs.webkit.org/show_bug.cgi?id=266539">https://bugs.webkit.org/show_bug.cgi?id=266539</a>

Reviewed by Aditya Keerthi.

At this point in its life ThemeMac is mainly responsible for
determining the sizing of a couple of form controls as the GPU process
has taken over most of the theming aspects.

And to determine the sizing of these form controls no cells are needed
so all that code can be removed.

And then we also remove some code that eventually does not do anything,
such as StyleAppearance::ColorWell which do to not having the correct
bezel style (which is itself a concept that can be removed) never ends
up being impacted by inflateControlPaintRect().

* Source/WebCore/platform/Theme.h:
(WebCore::Theme::inflateControlPaintRect const):
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::radioSize):
(WebCore::ThemeMac::inflateControlPaintRect const):
(WebCore::setControlSize): Deleted.
(WebCore::updateStates): Deleted.
(WebCore::configureToggleButton): Deleted.
(WebCore::createToggleButtonCell): Deleted.
(WebCore::sharedRadioCell): Deleted.
(WebCore::sharedCheckboxCell): Deleted.
(WebCore::buttonCell): Deleted.
(WebCore::setUpButtonCell): Deleted.
(WebCore::button): Deleted.
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustRepaintRect):

Canonical link: <a href="https://commits.webkit.org/272218@main">https://commits.webkit.org/272218@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f203488b2a2610ae188a160cbcfbcce400fa308

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30836 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9517 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32509 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33337 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27855 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31587 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11833 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6763 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27724 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31154 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7990 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27577 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6852 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7011 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27455 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34675 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28063 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27934 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33158 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7052 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5126 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30991 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8765 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7307 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7766 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7608 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/mac/ThemeMac.mm:
```diff
@@ -321,44 +281,21 @@ static LengthSize radioSize(const LengthSize& zoomedSize, float zoomFactor)
-    
-    if (appearance == StyleAppearance::Checkbox) {
-        [toggleButtonCell setButtonType:NSButtonTypeSwitch];
-        [toggleButtonCell setAllowsMixedState:YES];
```
It looks like `setAllowsMixedState` has been completely removed. But I think that's fine since it only has painting implications, which `ThemeMac` is not responsible for?

annevk replied:
That was my rationale for removing it, yeah. We just care about the size, that's it. All this extra code made it seem like more was going on.

annevk replied:
Oh, we can remove the cell completely. It just calls `controlSizeFromPixelSize()` like switch. Had not realized that the first time around. 🎉

annevk committed:
Remove cruft from ThemeMac
https://bugs.webkit.org/show_bug.cgi?id=266539

Reviewed by Aditya Keerthi.

At this point in its life ThemeMac is mainly responsible for
determining the sizing of a couple of form controls as the GPU process
has taken over most of the theming aspects.

And to determine the sizing of these form controls no cells are needed
so all that code can be removed.

And then we also remove some code that eventually does not do anything,
such as StyleAppearance::ColorWell which do to not having the correct
bezel style (which is itself a concept that can be removed) never ends
up being impacted by inflateControlPaintRect().

* Source/WebCore/platform/Theme.h:
(WebCore::Theme::inflateControlPaintRect const):
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::radioSize):
(WebCore::ThemeMac::inflateControlPaintRect const):
(WebCore::setControlSize): Deleted.
(WebCore::updateStates): Deleted.
(WebCore::configureToggleButton): Deleted.
(WebCore::createToggleButtonCell): Deleted.
(WebCore::sharedRadioCell): Deleted.
(WebCore::sharedCheckboxCell): Deleted.
(WebCore::buttonCell): Deleted.
(WebCore::setUpButtonCell): Deleted.
(WebCore::button): Deleted.
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustRepaintRect):

Canonical link: https://commits.webkit.org/272218@main
